### PR TITLE
[MU3] Sync CI scripts from master to 3.x

### PR DESCRIPTION
**Pull request by igorkorsukov**
Sync CI scripts from master to 3.x

### CODE REVIEW

File: .github/workflows/ci_windows.yml [hunk 1]
```diff
@@ -89,7 +86,7 @@ jobs:
       if: env.DO_APPCAST == 'true'   
       shell: bash
       run: |
-        bash ./build/ci/tools/sparkle_appcast_gen.sh -p windows   
+        bash ./build/ci/tools/sparkle_appcast_gen.sh -p windows
```
Comment (Jojo-Schmitz) [inline .github/workflows/ci_windows.yml:89]: still a trailing space left here

File: build/ci/linux/setup.sh [hunk 2]
```diff
@@ -136,12 +137,13 @@ cmake --version
 # POST INSTALL
 ##########################################################################
 
-chmod +x "${ENV_FILE}"
+chmod +x "$ENV_FILE"
```
Comment (Jojo-Schmitz) [inline build/ci/linux/setup.sh:140]: I believe the curly braces should rather stay, for it to not evailuate `$ENV` and append `"_FILE"`to it?

File: .github/workflows/ci_macos.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ jobs:
         echo "BUILD_MODE=$BUILD_MODE" >> $GITHUB_ENV
 
         DO_PUBLISH='false'
-        if [ "${{ github.event.inputs.publish }}" == "on" ]; then 
+        if [[ "${{ github.event.inputs.publish }}" == "on" || "$BUILD_MODE" == "nightly_build" ]]; then
```
Comment (Jojo-Schmitz) [inline .github/workflows/ci_macos.yml:42]: Probably doesn't matter, but this construct requires bash

File: build/ci/linux/setup.sh (hunk 2 above)
Comment (igorkorsukov) [reply build/ci/linux/setup.sh:140]: > I believe the curly braces should rather stay, for it to not evailuate `$ENV` and append `"_FILE"`to it?

I agree, I will return it

File: .github/workflows/ci_macos.yml (hunk 3 above)
Comment (igorkorsukov) [reply .github/workflows/ci_macos.yml:42]: On MacOS by default bash as shell

Comment (Jojo-Schmitz) [reply .github/workflows/ci_macos.yml:42]: Yes, I know. Just mentioning it, it is not a POSIX shell feature.

File: .github/workflows/ci_windows.yml (hunk 1 above)
Comment (igorkorsukov) [reply .github/workflows/ci_windows.yml:89]: removed

File: build/ci/linux/setup.sh (hunk 2 above)
Comment (igorkorsukov) [reply build/ci/linux/setup.sh:140]: returned

**Pull request merged**